### PR TITLE
fix(ci): Change the logic on junit upload job

### PR DESCRIPTION
**Pull request by chouetz**
First iterate on all xml to sort them by team
Then call upload on each team folder
Added tests
Muted all unittest that logged in stdout, pending a logger
implementation

### What does this PR do?
The initial logic was the following:
- Extract the tarball
- For each xml on it, split by owner
  - For each splitted xml, call uploader

As a result there was no parallelism at all leading to a duration of 20min to upload files for the [kmt_arm64_junit_upload](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Ajob%20%40ci.pipeline.name%3ADataDog%2Fdatadog-agent%20%40git.branch%3Amain%20%40ci.job.name%3Akmt_arm64_junit_upload&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=time%2Cdesc&colorBy=meta%5B%27ci.node.name%27%5D&colorByAttr=meta%5B%27ci.node.name%27%5D&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40ci.stage.name%2C%40ci.job.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch&currentTab=trace&fromUser=false&graphType=flamegraph&index=cipipeline&mode=sliding&saved-view-id=2643993&sort=time&spanID=886116639358716289&spanViewType=metadata&view=spans&viz=distribution&start=1716369150093&end=1716455550093&paused=false) job.
This PR changes the logic of the job:
- Extract the tarball
- For each xml, split by owner-flavor (for uniform tagging)
- Then, for each owner-flavor folder, call the uploader (with all the files at once) using a thread pool.

For the unit-tests jobs it moved from 102 calls to 31 only.
Before:
image
After:
image
For the specific `kmt_arm64_junit_upload` job
image

### Motivation
Job efficiency

### Additional Notes
I took advantage of this PR to

### CODE REVIEW

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by CelianR — APPROVED** [review]

File: tasks/libs/common/junit_upload_core.py [hunk 1]
```diff
@@ -70,50 +72,33 @@ def junit_upload_from_tgz(junit_tgz, codeowners_path=".github/CODEOWNERS"):
...
-        # for each unpacked xml file, split it and submit all parts
-        # NOTE: recursive=True is necessary for "**" to unpack into 0-n dirs, not just 1
-        xmls = 0
-        with collapsed_section("Uploading JUnit files"):
-            for xmlfile in glob.glob(f"{unpack_dir}/**/*.xml", recursive=True):
-                if not os.path.isfile(xmlfile):
+    junit_tgz = find_tarball(junit_tgz)
+
+    with collapsed_section(f"Uploading JUnit files for {junit_tgz}"):
+        with tempfile.TemporaryDirectory() as unpack_dir:
```
[the remark this answers is not in the document]
Comment (chouetz) [reply tasks/libs/common/junit_upload_core.py:78]: I'n not sure it will stand in a single line, checking

[the remark this answers is not in the document]
Comment (CelianR) [reply tasks/libs/common/junit_upload_core.py:78]: Yes indeed, as you wish

File: tasks/libs/common/junit_upload_core.py [hunk 2]
```diff
@@ -70,50 +72,33 @@ def junit_upload_from_tgz(junit_tgz, codeowners_path=".github/CODEOWNERS"):
...
+            working_dir = Path(unpack_dir)
+            # unpack all files from archive
+            with tarfile.open(junit_tgz) as tgz:
+                tgz.extractall(path=str(working_dir))
+
+            # Split xml files by codeowners
+            # Path().glob is recursive.
+            # We cast the glob generator to a list because we generate new xml files: that would lead to infinite loop
+            generated_xmls = 0
+            for xmlfile in list(working_dir.glob("**/*.xml")):
```
[the remark this answers is not in the document]
Comment (CelianR) [reply tasks/libs/common/junit_upload_core.py:88]: Oh ok I haven't paid attention, nice 👍

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by chouquette — APPROVED** [review]
LGTM for the BARX file

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**